### PR TITLE
Typewriter sound and customAudioclip

### PR DESCRIPTION
**Pull request by cnHeider**

### CODE REVIEW

**Comment by Spidersouris** [comment]
yes

File: lib/config-schema.coffee [hunk 1]
```diff
@@ -39,6 +39,23 @@ module.exports =
...
         default: true
 
+      audioclip:
+        title: "Play Audio - Audioclip"
+        description: "Which audio clip played at keystroke."
+        type: "string"
+        default: '../audioclips/gun.wav'
+        enum: [
+          {value: '../audioclips/gun.wav', description: 'Gun'}
+          {value: '../audioclips/typewriter-key-1.wav', description: 'Type Writer'}
```
Comment (lexcast) [inline lib/config-schema.coffee]: I hate to be so picky, but I don't want to add another audio, I think the custom audio option is enough for now.

File: lib/config-schema.coffee [hunk 2]
```diff
@@ -39,6 +39,23 @@ module.exports =
...
+          {value: 'customAudioclip', description: 'PathTo..'}
+        ]
+
+      customAudioclip:
+        title: "Play Audio - Path to Audioclip"
+        description: "Path to audioclip played at keystroke."
+        type: "string"
+        default: 'rocksmash.wav'
```
Comment (lexcast) [inline lib/config-schema.coffee]: `default` should be empty or removed. Maybe specify what formats are supported on the description.

File: lib/config-schema.coffee (hunk 1 above)
Comment (lexcast) [inline lib/config-schema.coffee]: Then the values should be: the first one `default` or `gun`, and the second `custom`.

File: lib/play-audio.coffee [hunk 3]
```diff
@@ -2,7 +2,10 @@ path = require "path"
 
 module.exports =
   play: ->
-    pathtoaudio = path.join(__dirname, '../audioclips/gun.wav')
+    if (@getConfig "audioclip") is "customAudioclip"
+      pathtoaudio = path.join(__dirname,"../audioclips/" + @getConfig "customAudioclip")
```
Comment (lexcast) [inline lib/play-audio.coffee]: Here the custom path for the audio should not be relative to the package folder. To not force the user to find this folder as it's easier to copy the audio clip path and paste it in the setting.

File: lib/play-audio.coffee [hunk 4]
```diff
@@ -2,7 +2,10 @@ path = require "path"
...
+    else
+      pathtoaudio = path.join(__dirname, @getConfig "audioclip");
     audio = new Audio(pathtoaudio);
```
Comment (lexcast) [inline lib/play-audio.coffee]: A little bit paranoid here, what happens when the audio is not found? 
Maybe a try/catch or a check with a fallback to the default audio clip.

**Comment by lexcast** [comment]
Thank you @cnHeider

**Pull request merged**